### PR TITLE
Fix nested sleep_ms calls and add tests

### PR DESCRIPTION
**Pull request by krleonid**
This PR fixes nested sleep_ms calls and adds comprehensive tests.

## Changes

- Fix nested sleep_ms calls
- Add tests for nested sleep functionality

## Testing

- Added test coverage for nested sleep scenarios

### CODE REVIEW

File: test/sql/function/generic/test_sleep.test [hunk 1]
```diff
@@ -136,6 +136,73 @@ DROP TABLE sleep_profiling_multi
...
 
+# Test nested sleep_ms calls - the inner sleep should execute
+# Test basic nested sleep_ms - should return NULL
+query T
+SELECT sleep_ms(sleep_ms(10))
+----
+NULL
+
+# Test nested sleep_ms with timing validation - inner sleep should execute
+# For nested sleep_ms(50) inside sleep_ms, we expect to see at least 0.04s (40ms) in the timing
```
Comment (maiadegraaf) [inline test/sql/function/generic/test_sleep.test]: Nit: shouldn't it be at least 0.05 (50ms)? I see you also test for >= 0.05 below.

**Review by maiadegraaf — COMMENTED** [review]
Thanks! Looks good, just one small nit.

Also since you added special null handling, could you just add tests with `sleep_ms(NULL)` and `sleep_ms(sleep_ms(NULL)`. And check that they don't sleep at all.

File: test/sql/function/generic/test_sleep.test (hunk 1 above)
Comment (krleonid) [reply test/sql/function/generic/test_sleep.test]: fixed

**Comment by krleonid** [comment]
@maiadegraaf 
I can’t check the upper bound, as it makes the tests flaky on some targets.
https://github.com/duckdb/duckdb/actions/runs/19350405757/job/55363388907

File: test/sql/function/generic/test_sleep.test [hunk 2]
```diff
@@ -136,6 +136,73 @@ DROP TABLE sleep_profiling_multi
...
+
+statement ok
+PRAGMA disable_profiling
+
+# Extract timing from JSON profiling output and validate it's >= 0.04s
+statement ok
+CREATE OR REPLACE TABLE sleep_nested_profiling AS SELECT * FROM read_json('__TEST_DIR__/sleep_nested_profiling.json')
+
+query T
+SELECT cpu_time >= 0.05 FROM sleep_nested_profiling WHERE contains(extra_info, 'sleep_ms') OR contains(query_name, 'sleep_ms') LIMIT 1
```
Comment (maiadegraaf) [inline test/sql/function/generic/test_sleep.test]: I'm just confused because in the comment you say you check if its `>= 0.04` but in the query you check `cpu_time >= 0.05`

**Review by maiadegraaf — APPROVED** [review]
Thanks for clearing that up! LGTM

**Comment by Mytherin** [comment]
Thanks!
